CMakeLists.txt

```diff
@@ -75,6 +75,9 @@ if(APPLE)
   include(MacRpath)
 endif()
 
+include(CheckIPOSupported)
+check_ipo_supported()
```

Comment:
Calling this without any arguments makes IPO being unsupported a fatal error. I guess we can either set `CMAKE_INTERPROCEDURAL_OPTIMIZATION` based on if IPO is supported, or only perform this check if `CMAKE_INTERPROCEDURAL_OPTIMIZATION` is true.